src/button_listener.c

```diff
@@ -3,41 +3,57 @@
 #include <signal.h>
 #include <gpiod.h>
 #include <stdarg.h>
-#include <curl/curl.h>
-#include <json-c/json.h>
 #include <string.h>
 #include <fcntl.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 
 #define BUTTON_LINE_NUMBER 80 // Black on GND and Red on GPIO
-//#define BUTTON_LINE_NUMBER 96
 #define DEBOUNCE_TIME 1000 // debounce time in milliseconds
-#define VOLTAGE_VALUE 1
+#define VOLTAGE_VALUE 1 // Voltage value corresponding to a button press being "on"
 
-// Time of the last button press
-static struct timespec last_release_time;
-// Whether the button is currently pressed
-static int button_pressed = 0;
+static struct timespec last_release_time; // Time of the last button press
+static int button_pressed = 0; // Flag for whether button is pressed
 
-const char *fifo_path = "/tmp/screenPipe";
+const char *fifo_path = "/tmp/screenPipe"; // Pipe file path
 
-struct gpiod_chip *chip;
-struct gpiod_line *button_line;
+struct gpiod_chip *chip; // GPIO chip struct
+struct gpiod_line *button_line; // GPIO line struct
 
-volatile sig_atomic_t interrupted = 0;
+volatile sig_atomic_t interrupted = 0; // Flag checking for whether the program was interrupted.
 
 void cleanup_gpio(void);
 
-// This function will be called when SIGINT is sent to the program (Ctrl+C)
 void handle_sigint(int sig) {
+    /**
```

Comment:
Use Doxygen-style to comment functions. Here is an example from GPT:
```c
/**
 * @brief Computes the area of a rectangle.
 *
 * Given the width and height of a rectangle, this function calculates and
 * returns its area.
 *
 * @param width The width of the rectangle. Must be non-negative.
 * @param height The height of the rectangle. Must be non-negative.
 * 
 * @return The area of the rectangle. Returns -1 if any parameter is negative.
 *
 * @note This function does not check for arithmetic overflow.
 */
double computeRectangleArea(double width, double height) {
    if (width < 0 || height < 0) {
        return -1.0;  // Error value
    }
    return width * height;
}
```
The `@brief` summarizes the function while `@note` are additional notes or warnings about the function (optional).